### PR TITLE
2.0 packaging

### PR DESCRIPTION
**Pull request by HarryDunham**
* Switch dockerfiles to leverage python 3.8.10
* Refactor `ggcore` and `ggsdk` into subdirectories of new directory `graphgrid_sdk`
* Refactor test imports away from flat `import ggcore` import scheme
* Add packaging resources (dockerfile, scripts)
* Change package name to remove `python`
* Add url link to doc homepage for pypi repo
* Append a1 release tag

### CODE REVIEW

File: package.dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+COPY pyproject.toml pyproject.toml
+COPY README.md README.md
+COPY scripts/in_container/build-and-upload-package build-and-upload-package
+COPY package_requirements.txt package_requirements.txt
+
+RUN pip3 install --upgrade pip && \
+    pip3 install -r package_requirements.txt
+
+ENTRYPOINT [ "sh", "build-and-upload-package" ]
+CMD []
```
Comment (alexiudice) [inline package.dockerfile:16]: @HarryDunham Out of curiosity what is this empty `CMD []` line providing?

File: package_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+build==0.7.0
+twine==4.0.0
```
Comment (alexiudice) [inline package_requirements.txt:2]: is this replacing the current `requirements.txt`? Do we need to move over all the dependencies originally in there?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools>=57"]
```
Comment (alexiudice) [inline pyproject.toml:2]: similarly to my comment on `package_requirements`, is this replacing `requirements.txt`? Do we need to move over the dependencies originally in that file?

**Review by alexiudice — COMMENTED** [review]
Left a few questions, overall looks good and the refactors of the project structure make sense to me :+1:

File: package_requirements.txt (hunk 2 above)
Comment (HarryDunham) [reply package_requirements.txt:2]: No, these are simply the requirements in order to build the sdk package and upload it to pypi. Nothing else is required

File: pyproject.toml (hunk 3 above)
Comment (HarryDunham) [reply pyproject.toml:2]: No, please refer to PEP 508 [here](https://peps.python.org/pep-0518/#file-format) for a more indepth breakdown

**Review by alexiudice — APPROVED** [review]

File: package.dockerfile (hunk 1 above)
Comment (HarryDunham) [reply package.dockerfile:16]: Functionally nothing, it was entirely stylistic to mark the need to overwrite/provide proper commands to the container on run

**Review by NMPowell99 — APPROVED** [review]

**Review by bradnussbaum — APPROVED** [review]

**Pull request merged**